containers/sip-servlets-as7/src/main/java/org/jboss/as/clustering/web/infinispan/sip/DistributedCacheManager.java

```diff
@@ -0,0 +1,1302 @@
+		
+		trace("removeSipApplicationSession(%s)", sipAppSessionId);
+		
+		this.removeSipApplicationSession(sipAppSessionId, false);
+	}
+	
+	private void removeSipApplicationSession(final String sipAppSessionId, final boolean local) {
+		if (logger.isDebugEnabled()){
+			logger.debug("removeSipApplicationSession - sipAppSessionId=" + sipAppSessionId + ", local=" + local);
+		}
+		trace("removeSipApplicationSession(%s, %s)", sipAppSessionId, local);
+		
+		final String sipAppSessionKey = getSipAppSessionCacheKey(sipAppSessionId);
+		
+		Operation<Void> operation = new Operation<Void>() {
+            @Override
+            public Void invoke(Cache<String, Map<Object, Object>> cache) {
+                cache.remove(sipAppSessionKey);
+                // TODO: also remove sip sessions of app session?
+                return null;
+            }
+        };
+        getInvoker().invoke(getCache(), operation, Flag.SKIP_CACHE_LOAD, local ? Flag.CACHE_MODE_LOCAL : Flag.SKIP_REMOTE_LOOKUP);
+    }
+
+	@Override
+	public void removeSipSession(String sipAppSessionId, String sipSessionId) {
+		if (logger.isDebugEnabled()){
+			logger.debug("removeSipSession - sipAppSessionId=" + sipAppSessionId + ", sipSessionId=" + sipSessionId);
+		}
+		
+		trace("removeSipSession(%s, %s)", sipAppSessionId, sipSessionId);
+		
+		this.removeSipSession(sipAppSessionId, sipSessionId, false);
+	}
+
+	private void removeSipSession(final String sipAppSessionId, final String sipSessionId, final boolean local) {
+		if (logger.isDebugEnabled()){
+			logger.debug("removeSipSession - sipAppSessionId=" + sipAppSessionId + ", sipSessionId=" + sipSessionId + ", local=" + local);
+		}
+		trace("removeSipSession(%s, %s, %s)", sipAppSessionId, sipSessionId, local);
+		
+		final String sipSessionKey = getSipSessionCacheKey(sipAppSessionId, sipSessionId);
+		
+		Operation<Void> operation = new Operation<Void>() {
+            @Override
+            public Void invoke(Cache<String, Map<Object, Object>> cache) {
+            	cache.remove(sipSessionKey);
+                
+                return null;
+            }
+        };
+        getInvoker().invoke(getCache(), operation, Flag.SKIP_CACHE_LOAD, local ? Flag.CACHE_MODE_LOCAL : Flag.SKIP_REMOTE_LOOKUP);
+    }
+	
+	@Override
+	public void removeSipApplicationSessionLocal(String sipAppSessionId) {
+		if (logger.isDebugEnabled()){
+			logger.debug("removeSipApplicationSessionLocal - sipAppSessionId=" + sipAppSessionId);
+		}
+		trace("removeSipApplicationSessionLocal(%s)", sipAppSessionId);
+		
+        this.removeSipApplicationSession(sipAppSessionId, true);
+	}
+	
+	@Override
+	public void removeSipApplicationSessionLocal(String sipAppSessionId, String dataOwner) {
+		if (logger.isDebugEnabled()){
+			logger.debug("removeSipApplicationSessionLocal - sipAppSessionId=" + sipAppSessionId + ", dataOwner=" + dataOwner);
+		}
+		
+		trace("removeSipApplicationSessionLocal(%s, %s)", sipAppSessionId, dataOwner);
+		
+		if (dataOwner == null) {
```

Comment:
This is also derived from the corresponding web replication related JBoss class (org.jboss.as.clustering.web.infinispan.DistributedCacheManager) and works the same way as in the JBoss codes.